example/src/modules/user.ts

```diff
@@ -23,7 +30,20 @@ const getDefaultUser = (): IUser => ({
   name: '',
 })
 
-export const userModule = new Module<IUser, IUserGetters, IUserMutations>({
+const localStoragePlugin: ILastPlugin<IUser> = {
+  onStateInit: (state) => {
+    const stringifiedState = localStorage.getItem('user')
+    if (!stringifiedState) {
+      return state as IUser
+    }
+    return JSON.parse(stringifiedState) as IUser
+  },
+  onDataChange: (value) => {
+    localStorage.setItem('user', JSON.stringify(value))
+  },
+}
```

Comment:
If the plugin is the only plugin applies to a module, it has to be of type `ILastPlugin<T>`, so that the module is guaranteed to receive a value of the same type as the initial state.
But since the `state` argument in `onStateInit` is `Partial<T>`, we have to explicitly cast it to `<T>` to pass the type check, which I think is not ideal.
Is there a recommended solution to this?

Reply:
Besides, this is not good for making plugins into libraries as well.
Because we have no idea if the plugin will be the last one or not, so developers will have to export two version, `IPlugin<T>` and `ILastPlugint<T>`, this is very unintuitive (is it a word?).

Reply:
I made the plugins a little more restrictive to solve this problem. Changing the init signature to `onStateInit?: (state: S) => S` means we can't have intermediate `Partial<S>` in the plugin system, but I think it's worth it. A plugin author can cast the return value if they really want to, and I don't think there are many situations where it's needed.

I also think starting more restrictive and relaxing requirements later is better than the other way around, it's less likely to break published code.